lib/mongoid/clients/sessions.rb

```diff
@@ -44,6 +44,12 @@ def with_session(options = {})
           else
             raise ex
           end
+        rescue Mongo::Error::OperationFailure => ex
+          if ex.server_message =~ /startTransaction' is an unknown field/
```

Comment:
Can we change this to check for the error code?

Reply:
Done